`Chore`: Set tum artemis server as default

FelberMartin opened the pull request:
<!-- Feel free to leave out sections that are not appropriate for your PR.  -->

### Problem Description
The current Artemis Android release in the Play Store uses the "Tum" build flavor, which does not allow instance (artemis server) selection. Simply switching the released build flavor to "unrestricted" would have caused already logged in users with weird behaviour (all network calls would fail because it would try to reach the artemis server via localhost; only a logout and sign in again would have resolved this issue).

### Changes
<!-- Descripe your changes on a high level. If you feel like technical details would be helpful for the reviewer, please add them here. --> 
- Changed the default server for the "unrestricted" flavor to the tum artemis server
- Adapted settings code to rely on the `serverUrl` instead of the duplicated logic with `hasUserSelectedInstance`
  - Removed now no longer needed code `hasUserSelectedInstance`
- Fixed bug introduced by https://github.com/ls1intum/artemis-android/pull/222, as it removed the `NestedDestination.InstanceSelection` navigation target, but did not remove it in other places of the code. This would lead to an app crash just after freshly installing the app.

### Steps for testing
#### Scenario 1 - Updating the app from "tum" to "unrestricted" flavor
- Uninstall the app on the test device
- Switch the build variant by Menu > Build > Select Build variant ... and select "productionTumDebug" for `:app`
- Launch the app and log in with your TUM credentials
- Now we check that changing to new release version with the "unrestricted" flavor works as expected:
  - Change the build variant to "productionUnrestrictedDebug" for `:app`
  - Launch the app
  - See that you are still logged in with your account to the TUM server and the dashboard is loaded

#### Scenario 2 - Fresh install
- Uninstall the app on the test device
- Directly change the build variant to "productionUnrestrictedDebug" for `:app`
- Launch the app
- See that by default the tum instance is selected for login, but the user can also select a different server by pressing "not your university"

eylulnc commented:
Both scenario works as expected, I only have few suggestions,

1 - Maybe we can open the instance bottom sheet as full
2 - In iOS if we select different instance that default,  itshows which one maybe we can also have sth like this

<img width="372" alt="Screenshot 2024-12-30 at 12 28 25" src="https://github.com/user-attachments/assets/67dd0e72-0c93-4a95-94b2-a9f4e0a97f34" />

eylulnc reviewed (approved):
Lgtm!

FelberMartin commented:
I addressed your comments and the BottomSheet now opens in full screen. 
I also added an indicator of the the server host, however I thought it mainly is helpful for us developers (which test server do we have selected?), but not really necessary for the end users, so I decided to only show it in the debug version. Do you think this is sufficient @eylulnc ?

julian-wls reviewed (approved):
Just tested your changes and I did not notice any issues. The code changes seem appropriate.
